iothub/service/src/Jobs/ScheduledJobsClient.cs

```diff
@@ -0,0 +1,364 @@
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided <paramref name="cancellationToken"/> has requested cancellation.</exception>
+        public virtual Task<JobResponse> ScheduleTwinUpdateAsync(ScheduledTwinUpdate scheduledTwinUpdate, ScheduledJobsOptions scheduledJobsOptions = default, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"queryCondition=[{scheduledTwinUpdate.QueryCondition}]", nameof(ScheduleDeviceMethodAsync));
+            try
+            {
+                Argument.RequireNotNull(scheduledTwinUpdate, nameof(scheduledTwinUpdate));
+                Argument.RequireNotNullOrEmpty(scheduledTwinUpdate.QueryCondition, nameof(scheduledTwinUpdate.QueryCondition));
+                Argument.RequireNotNull(scheduledTwinUpdate.Twin, nameof(scheduledTwinUpdate.Twin));
+                Argument.RequireNotNull(scheduledTwinUpdate.StartTimeUtc, nameof(scheduledTwinUpdate.StartTimeUtc));
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var jobRequest = new JobRequest
+                {
+                    JobId = string.IsNullOrWhiteSpace(scheduledJobsOptions.JobId) ? "job_" + Guid.NewGuid().ToString() : scheduledJobsOptions.JobId,
+                    JobType = JobType.ScheduleUpdateTwin,
+                    UpdateTwin = scheduledTwinUpdate.Twin,
+                    QueryCondition = scheduledTwinUpdate.QueryCondition,
+                    StartTime = scheduledTwinUpdate.StartTimeUtc,
+                    MaxExecutionTimeInSeconds = scheduledJobsOptions.MaxExecutionTimeInSeconds
+                };
+
+                return CreateAsync(jobRequest, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(ScheduleTwinUpdateAsync)} threw an exception: {ex}", nameof(ScheduleTwinUpdateAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Enter(this, $"queryCondition=[{scheduledTwinUpdate.QueryCondition}]", nameof(ScheduleDeviceMethodAsync));
+            }
+
+        }
+
+        private async Task<JobResponse> CreateAsync(JobRequest jobRequest, CancellationToken cancellationToken)
```

Comment:
Having a private function to house what is basically 4 lines of code seems unnecessary to me. Especially since it is only being called from 2 places. I'd move the HTTP operations into the functions that call this one and delete this private function